app/Imports/BillingDataImport.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace App\Imports;
+
+use Illuminate\Http\UploadedFile;
+use PhpOffice\PhpSpreadsheet\Reader\Csv;
+use App\Imports\Filters\BillingDataReadFilter;
+use Carbon\Carbon;
+
+class BillingDataImport
+{
+    /**
+     * @param int $billing_id
+     * @param UploadedFile $file
+     * @return array $preparedBillingData
+     */
+    public function getBillingDataToImport($billing_id, $file): array
```

Comment:
Gdy zrobisz tak jak opisałem w kontrolerze to znika Ci wyciek biznesowej wiedzy o array. (Kontrolera nie powinno obchodzić, że to jakiś specjalny array).  Tutaj jest też błąd. Dałeś mi cynk, że ten Array może być duży. Skoro na tym etapie rozwoju jest to prosta aplikacja to lepiej przekazywać ten array przez referencję - unikasz wten czas kopiowania danych (które się dzieje w przypadku tablic i przekazywania danych do/z metody).

Reply:
Nie do końca rozumiem jak mam tutaj użyć tej referencji. Stykam się z tym pierwszy raz i poczytałem trochę + zobaczyłem przykłady i z tego co tam widzę to zmienną referencyjną można oznaczyć argument funkcji albo zmienną. Ja tutaj tej tablicy nie przekazuje nigdzie do innej funkcji tylko ją zwracam. A jak próbuję dać ją przed odwołaniem się do funkcji to mam Only variables should be assigned by reference